cachi2/core/package_managers/yarn_classic/main.py

```diff
@@ -27,6 +27,26 @@
 
 
 MIRROR_DIR = "deps/yarn-classic"
+_yarn_classic_pattern = re.compile("yarn lockfile v1")  # See [yarn_classic_trait].
+
+
+class MissingLockfile(PackageRejected):
+    """Indicate that a lock file is missing."""
+
+    def __init__(self) -> None:
+        """Initialize a Missing Lockfile error."""
+        reason = "Yarn lockfile 'yarn.lock' missing, refusing to continue"
+        solution = "Make sure your repository has a Yarn lockfile (i.e. yarn.lock) checked in"
+        super().__init__(reason, solution=solution)
+
+
+class NotV1Lockfile(PackageRejected):
+    """Indicate that a lockfile is of wrong tyrpoe."""
+
+    def __init__(self, package_path: Any) -> None:
+        """Initialize a Missing Lockfile error."""
+        reason = f"{package_path} not a Yarn v1"
+        super().__init__(reason, solution=None)
```

Comment:
Recently, I got a comment from Erik, that all custom cachi2 exceptions should be inside `cachi2.core.errors.py`. But that's just a nitpick.

Reply:
Correction - I meant any publicly exposed top-level exceptions like `PathOutsideRoot`, `PackageRejected`, etc. ...custom exceptions which are module-scoped are fine to keep in the given module where it makes the most sense for them. That said, these **are** module-scoped exceptions and so we should start their name with an underscore. Additionally, these should be re-raised in the `fetch-yarn-...` meta entrypoint as a `PackageRejected` exception IMO as that's the public-facing one.

Reply:
These are not module-scoped, but they are also not cachi2-scoped: they are raised here to send a message to meta-PM. They do not need to be re-wrapped because as we have agreed earlier everything that is not v1 is v2+ (and exceptions from v2+ propagate freely). Furthermore, any other exception will propagate to the user, so if something happens to conform to v1 and not trigger any of these, and also is a broken v1 then a user would know.